src/proof/drat/drat_proof.cpp

```diff
@@ -0,0 +1,235 @@
+/*********************                                                        */
+/*! \file drat_proof.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief DRAT Proof Format
+ **
+ ** Defines deserialization for DRAT proofs.
+ **/
+
+#include "proof/drat/drat_proof.h"
+#include <bitset>
+
+namespace CVC4 {
+namespace proof {
+namespace drat {
+
+// Forward-declare helper functions for parsing the binary DRAT format.
```

Comment:
Could you please [justify](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#forward-declarations) this forward declaration.

Reply:
Good catch! I don't have a good justification. I'll lift the implementations to this point in the source.